tests/workflow_tests/test_events.py

```diff
@@ -63,17 +63,30 @@ def test_sorted_events(self):
         self.assertListEqual(events.items, sorted_events)
 
     def test_filtered_events(self):
-        deployment_id = self.deployment_id
-        # create another deployment to test correct filtering
-        self._create_deployment()
-        filters = {'deployment_id': deployment_id}
+        # create multiple deployments
+        deployment_ids = []
+        for i in range(3):
+            deployment_ids.append(self._create_deployment())
+
+        # filter a subset of the deployments
+        expected_deployment_ids = deployment_ids[:2]
+        filters = {'deployment_id': expected_deployment_ids}
         events = self.client.events.list(**filters)
+
         self.assertGreater(len(events), 0, 'No events')
+
+        # count events for each requested deployment id
+        events_per_deployment = \
```

Comment:
how about instead:
deployments_with_events = {event['context']['deployment_id'] for event in events}
self.assertEquals(deployments_with_events, set(expected_deployment_ids))

seems shorter
also, if at some point we break the filter and somehow the third deployment also returns an event, the current code will break on line 83